packages/js/src/components/social/useFallbackWarning.js

```diff
@@ -0,0 +1,31 @@
+import { useEffect, useState } from "@wordpress/element";
+import { __, sprintf } from "@wordpress/i18n";
+
+/**
+ * Checks if the fallback image is in AVIF format and sets a warning message.
+ *
+ * @param {string} imageFallbackUrl
+ * @param {string} imageUrl
+ * @param {array} imageWarnings
+ *
+ * @returns {array}
+ */
+export const useFallbackWarning = ( imageFallbackUrl, imageUrl, imageWarnings ) => {
+	const [ hasFallbackWarning, setHasFallbackWarning ] = useState( false );
+	const warningMessage = sprintf(
+		/* Translators: %s expands to the jpg format, %s expands to the png format, %s expands to the webp format, %s expands to the gif format. */
+		__(
+			"The image automatically added from your content isn't supported. Please use %s, %s, %s or %s formats to ensure it displays correctly on social media.",
+			"wordpress-seo"
+		),
+		"JPG", "PNG", "WEBP", "GIF"
+	);
+	useEffect( () => {
+		if ( imageUrl === "" ) {
+			setHasFallbackWarning(  imageFallbackUrl.toLowerCase().endsWith( ".avif" )  );
+		} else {
+			setHasFallbackWarning( false );
+		}
```

Comment:
Extra spacing inside the `setHasFallbackWarning` there. But also if/else replaced by a ternary would look like:

```suggestion
		setHasFallbackWarning( imageUrl === "" ? imageFallbackUrl.toLowerCase().endsWith( ".avif" ) : false );
```